motionplan/motionPlanner.go

```diff
@@ -385,3 +382,90 @@ IK:
 	}
 	return orderedSolutions, nil
 }
+
+// CheckPlan checks if obstacles intersect the trajectory of the frame following the plan.
+func CheckPlan(
+	checkFrame frame.Frame,
+	plan []map[string][]frame.Input,
+	worldState *frame.WorldState,
+	fs frame.FrameSystem,
+	errorState spatialmath.Pose,
+) error {
+	// since there are only two states we care about, lets just return err
+	// ensure that we can actually perform the check
+	if len(plan) < 2 {
+		return errors.New("plan must have at least two elements")
+	}
+
+	// construct solverFrame
+	// Note that this requires all frames which move as part of the plan, to have an
+	// entry in the very first plan waypoint
+	sf, err := newSolverFrame(fs, checkFrame.Name(), frame.World, plan[0])
+	if err != nil {
+		return err
+	}
+
+	// construct planager
+	sfPlanner, err := newPlanManager(sf, fs, golog.NewLogger("checkPlan-logger"), defaultRandomSeed)
+	if err != nil {
+		return err
+	}
+
+	// convert plan into nodes
+	planNodes := make([]node, 0, len(plan))
+	for _, step := range plan {
+		stepConfig, err := sf.mapToSlice(step)
+		if err != nil {
+			return err
+		}
+		pose, err := sf.Transform(stepConfig)
+		// adjust pose based off how much we've deviated from the expected path
+		pose = spatialmath.Compose(pose, errorState)
+		if err != nil {
+			return err
+		}
+		planNodes = append(planNodes, &basicNode{q: stepConfig, pose: pose})
+	}
+
+	// This should be done for any plan whose configurations are specified in relative terms rather than absolute ones.
+	// Currently this is only TP-space, so we check if the PTG length is >0.
+	// The solver frame will have had its PTGs filled in the newPlanManager() call, if applicable.
+	relative := len(sf.PTGs()) > 0
+
+	if relative {
+		if planNodes, err = rectifyTPspacePath(planNodes, sf); err != nil {
```

Comment:
If the `plan` passed in is only the remaining nodes, then we'll also need to pass in the current nominal position of the robot along the plan. `rectifyTPspacePath` will return a set of plan nodes such that each node is transformed from the zero pose, so we need to know where the first node in the plan ought to be.

Reply:
Just want to follow up on this, has this been implemented yet?